<launch>

  <arg name="model" default="$(find wpb_home_bringup)/urdf/wpb_home.urdf"/>
  <arg name="gui" default="false" />
  <arg name="rvizconfig" default="$(find rushbot)/rviz/rushbot_main.rviz" />

  <param name="robot_description" command="$(find xacro)/xacro.py $(arg model)" />
  <param name="use_gui" value="$(arg gui)"/>
  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" >
    <rosparam command="load" file="$(find wpr_simulation)/config/wpb_home.yaml" />
  </node>

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher"/>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />
<!-- ##################################################################################### -->

  <!-- Run the map server -->
  <node name="map_server" pkg="map_server" type="map_server" args="$(find rushbot)/data/map.yaml"/>

  <!-- Map tools  -->
  <node pkg="waterplus_map_tools" type="wp_manager" name="wp_manager">
    <param name="load" type="string" value="$(find rushbot)/data/waypoints.xml"/> 
  </node>

  <!--- Run AMCL -->
  <include file="$(find wpb_home_bringup)/nav_lidar/amcl_omni.launch" />

  <!--- Run move base -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base">
    <rosparam file="$(find wpb_home_bringup)/nav_lidar/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find wpb_home_bringup)/nav_lidar/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find wpb_home_bringup)/nav_lidar/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find wpb_home_bringup)/nav_lidar/global_costmap_params.yaml" command="load" />
    <param name="base_global_planner" value="global_planner/GlobalPlanner" /> 
    <param name="use_dijkstra" value="true"/>
    <param name="base_local_planner" value="wpbh_local_planner/WpbhLocalPlanner" />

    <param name= "controller_frequency" value="10" type="double"/>
  </node>

  <!-- Run script -->
  <include file="$(find rushbot)/launch/rushbot_grab.launch" />
  
  <node pkg="rushbot" type="rushbot_main" name="rushbot_main" output="screen"/>

</launch>
